src/nspawn/nspawn.c

```diff
@@ -1018,14 +1011,17 @@ static int parse_argv(int argc, char *argv[]) {
                         assert_not_reached("Unhandled option");
                 }
 
+        if (getenv_bool("SYSTEMD_NSPAWN_SHARE_SYSTEM") > 0)
+                arg_share_system = true;
```

Comment:
Frankly, I don't see the advantage of removing the option (from actual parsing, not documentation), to add an environment variable with the same functionality. I haven't actually used `--share-system` myself, so I don't have a very strong opinion, but it *was* documented, and people might be using it, so it seems nicer to follow the normal deprecation process.